src/components/H/H.tsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import Box, { BoxProps } from '../Box';
+
+const DEFAULT_HEADING_LEVEL = 2;
+export const HeadingLevelContext = React.createContext(DEFAULT_HEADING_LEVEL);
+
+export interface HProps extends BoxProps<'h2'> {
+  /**
+   * Allows you to manually tweak the rendered heading level above the heading level's
+   * context.
+   *
+   * This can be useful if you want to render two heading levels right next to each other without
+   * needing an intermediate <Section />.
+   *
+   * @example
+   * ```tsx
+   * <H>this is a h2</H>
+   * <H increase={1}>this is a h3</H>
+   * <H increase={2}>this is a h4</H>
+   * ```
+   *
+   * You can only increase up to 4 levels, since <H> only renders <h2>-<h6> tags
+   */
+  increase?: 1 | 2 | 3 | 4;
+}
+
+type HTag = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
+
+/**
+ * Provides and manages `h` tags within a document context. The underlying heading level rendered
+ * can vary depending on _where_ this component is in the component tree.
+ *
+ * This allows for simple scaffolding of a document with accessible heading levels while
+ * also allowing components to create markup in a vacuum.
+ */
+export const H: React.FC<HProps> = React.forwardRef<HTMLHeadingElement, HProps>(function H(
+  { increase = 0, ...rest },
```

Comment:
I don't think naming the prop `level` portrays the right mental model—not if we want these tags to automatically manage levels properly. The primary reason being that `level` implies the absolute level, rather than the rest of the component operating with a _relative awareness_ of its level.

```tsx
// Let's assume we're at the h4 level in the ctx.
// I'd expect this to output an `h2` by a simple reading of the component.
<H level={2}>Heading</H>
```

I wonder if we can axe `level` or `increase` entirely? The situations where we would need two adjacent heading levels are decently rare, especially since we don't have SEO as a main concern. In the off-chance that we do need adjacent headings, you can always wrap the heading in a new section to increase the level.

```tsx
<H>Heading 2</H>
<Section as='span'>
  <H>Heading 3</H>
</Section>
```